asm and std=c99 incompatibility

Calls to extended `asm` fail to compile with gcc when `std=c99` compiler flag is set

### Example
```nim
{.passC: "-std=c99".}

proc main =
  let src = 41
  var dst = 0

  asm """
    mov %1, %0\n\t
    add $1, %0
    : "=r" (`dst`)
    : "r" (`src`)"""

  echo dst

main()
```

### Current Output
```
mtest.nim.c:104:5: error: expected ')'
    : "=r" (dst)
    ^
mtest.nim.c:102:5: note: to match this '('
        asm("    mov %1, %0\n\t\n"
           ^
1 error generated.
```

### Expected Output
```
42
```

### Possible Solution

I have a PR with the following change and a test, however this definitely feels out of my depth

In `extccomp.nim`, under `compiler gcc:`, replace

```nim
asmStmtFrmt: "asm($1);$n",
```

with 

```nim
asmStmtFrmt: "__asm__($1);$n",
```

```
$ nim -v

Nim Compiler Version 1.7.1 [MacOSX: amd64]
Compiled at 2022-07-10
Copyright (c) 2006-2022 by Andreas Rumpf

git hash: e8ee2f9c2ad06cad2f62fe7505acbb43530f28d7
```

--- a/compiler/extccomp.nim
+++ b/compiler/extccomp.nim
@@ -86,7 +86,7 @@ compiler gcc:
     linkLibCmd: " -l$1",
     debug: "",
     pic: "-fPIC",
-    asmStmtFrmt: "asm($1);$n",
+    asmStmtFrmt: "__asm__($1);$n",
     structStmtFmt: "$1 $3 $2 ", # struct|union [packed] $name
     produceAsm: gnuAsmListing,
     cppXsupport: "-std=gnu++14 -funsigned-char",

--- a/tests/compiler/tasm.nim
+++ b/tests/compiler/tasm.nim
@@ -1,0 +1,14 @@
+{.passC: "-std=c99".}
+
+block asmTest:
+  when defined(gcc):
+    let src = 41
+    var dst = 0
+
+    asm """
+      mov %1, %0\n\t
+      add $1, %0
+      : "=r" (`dst`)
+      : "r" (`src`)"""
+
+    doAssert dst == 42

--- a/tests/compiler/tasm.nim
+++ b/tests/compiler/tasm.nim
@@ -1,14 +1,15 @@
-{.passC: "-std=c99".}
+proc testAsm() =
+  let src = 41
+  var dst = 0
 
-block asmTest:
-  when defined(gcc):
-    let src = 41
-    var dst = 0
+  asm """
+    mov %1, %0\n\t
+    add $1, %0
+    : "=r" (`dst`)
+    : "r" (`src`)"""
 
-    asm """
-      mov %1, %0\n\t
-      add $1, %0
-      : "=r" (`dst`)
-      : "r" (`src`)"""
+  doAssert dst == 42
 
-    doAssert dst == 42
+when defined(gcc) or defined(clang) and not defined(cpp):
+  {.passc: "-std=c99".}
+  testAsm()

